alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/plugins/api/ToggleableHealthProcessorPlugin.java

```diff
@@ -0,0 +1,28 @@
+package eu.xenit.alfresco.healthprocessor.plugins.api;
+
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthReport;
+import java.util.Set;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.slf4j.Logger;
+
+@Slf4j
+@Data
+public abstract class ToggleableHealthProcessorPlugin implements HealthProcessorPlugin {
```

Comment:
Can this be an interface that specializes in the `enabled` part?

Reply:
I think it does specialize in the enabled part? However since it contains an `enabled` field, it can be at most an `abstract class`?

Reply:
`HealthProcessorPlugin` has `isEnabled` as well, hmm :thinking:

Reply:
What would the use of this class be?
To set the `private boolean enabled` to another value?

Reply:
This class is a tool that can be used to easily support enabling / disabling via a single filed, which easily translates to an Alfresco property in the XML context. 

Other implementations of `HealthProcessorPlugin` can individually decide if they are are enabled or not based on whatever they want. (A certain state in / of the repository for example?)

Reply:
Okay

At some point we can use composition for that instead of inheritance (for example have some sort of a reusable AlfrescoPropertyEnabledChecker component that is instantiated with the key to check in the xml)
But let's go with this class for now :+1:

Reply:
Composition over inheritance is indeed always a good idea. 

Brainstorming about how that would be possible here. Suppose we have a `AlfrescoPropertyEnabledChecker`, we would still need to inject it in the `HealthProcessorPlugin` implementation, right? 

Ofcourse you can say that the implementation needs to do that, however in that case I would prolly start making another abstract class that injects the property checker and handles the `isEnabled()` method. 

Thoughts?

Reply:
We could combine that with a Creational pattern that knows how to compose which plugins with which Checkers